packages/wallet/src/test-utils/launchNode.ts

```diff
@@ -0,0 +1,172 @@
+    const graphQLStartSubstring = 'Binding GraphQL provider to';
+
+    const command = useSystemFuelCore ? 'fuel-core' : './node_modules/.bin/fuels-core';
+
+    const ipToUse = ip || '0.0.0.0';
+
+    const portToUse =
+      port ||
+      (
+        await getPortPromise({
+          port: 4000, // tries 4000 first, then 4001, then 4002, etc.
+          stopPort: 5000, // don't try ports above 5000
+        })
+      ).toString();
+
+    const child = spawn(command, [
+      'run',
+      '--ip',
+      ipToUse,
+      '--port',
+      portToUse,
+      '--db-type',
+      'in-memory',
+      '--consensus-key',
+      consensusKey,
+      '--chain',
+      chainConfigPath,
+      ...args,
+    ]);
+
+    // Cleanup function where fuel-core is stopped.
+    const cleanup = () => {
+      kill(Number(child.pid));
+
+      // Remove all the listeners we've added.
+      child.stdout.removeAllListeners();
+      child.stderr.removeAllListeners();
+    };
+
+    child.stderr.setEncoding('utf8');
+
+    // Look for a specific graphql start point in the output.
+    child.stderr.on('data', (chunk: string) => {
+      // Look for the graphql service start.
+      if (chunk.indexOf(graphQLStartSubstring) !== -1) {
+        // Resolve with the cleanup method.
+        resolve({
+          cleanup,
+          ip: ipToUse,
+          port: portToUse,
+        });
+      }
+    });
+
+    // Process exit.
+    process.on('exit', cleanup);
+
+    // Catches ctrl+c event.
+    process.on('SIGINT', cleanup);
+
+    // Catches "kill pid" (for example: nodemon restart).
+    process.on('SIGUSR1', cleanup);
+    process.on('SIGUSR2', cleanup);
+
+    // Catches uncaught exceptions.
+    process.on('uncaughtException', cleanup);
+  });
+
+const generateWallets = async (count: number, provider: Provider) => {
+  const wallets: WalletUnlocked[] = [];
+  for (let i = 0; i < count; i += 1) {
+    const wallet = await generateTestWallet(provider, [[1_000, BaseAssetId]]);
+    wallets.push(wallet);
+  }
+  return wallets;
+};
+
+/**
+ * Launches a fuel-core node and returns a provider, 10 wallets, and a cleanup function to stop the node.
+ * @param launchNodeOptions - options to launch the fuel-core node with.
+ * @param walletCount - the number of wallets to generate. (optional, defaults to 10)
+ * */
+export const launchNodeAndGetWallets = async ({
+  launchNodeOptions,
+  walletCount = 10,
+}: {
+  launchNodeOptions?: Partial<LaunchNodeOptions>;
+  walletCount?: number;
+} = {}) => {
+  const osTempDir = os.tmpdir();
+
+  const subDirName = '.fuels-ts'; // Change to your desired subfolder name
+  const subDirPath = path.join(osTempDir, subDirName);
+
+  fsSync.mkdirSync(subDirPath, { recursive: true });
+
+  const chainConfigFilePath = path.join(subDirPath, '.chainConfig.json');
+
+  // Write a temporary chain configuration file.
+  await fs.writeFile(chainConfigFilePath, JSON.stringify(defaultChainConfig), 'utf8');
```

Comment:
It would be good to allow chainConfig to be provided by the user. And get the consensusKey from a `Wallet.generate()` or a user-provided one.

These things will help tests to have specific chain states before the tests, like particular messages.

Reply:
@luizstacio you can already do both of these things by passing in the respective `launchNodeOptions` like so:

const { stop, provider, wallets } = await launchNodeAndGetWallets({
    launchNodeOptions: {
      chainConfigPath: './.fuel-core/configs/chainConfig.json',
      consensusKey: 'key'
    },
  });

Reply:
We only use the default chain config and the default consensus key in case the user hasn't provided one. I did some cleanup though. Please have a look! :)